internal/controller/sparkapplication/controller.go

```diff
@@ -172,6 +172,7 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		return ctrl.Result{Requeue: true}, err
 	}
 	logger.Info("Reconciling SparkApplication", "name", app.Name, "namespace", app.Namespace, "state", app.Status.AppState.State)
+	defer logger.Info("Finished reconciling SparkApplication", "name", app.Name, "namespace", app.Namespace)
```

Comment:
```suggestion
	defer logger.Info("Finished reconciling SparkApplication", "name", app.Name, "namespace", app.Namespace, "state", app.Status.AppState.State)
```

We can add info about application state in the log statement.

Reply:
I actually tried this but it doesn't work because the `app` object we use here is not the same one that we mutate during reconcile. I think we would need to call `getSparkApplication` again so I thought it probably wasn't worth it. Without status this log line's value is reduced, but I decided it was probably worth keeping.